eng/common/scripts/Delete-RemoteBranches.ps1

```diff
@@ -50,9 +83,20 @@ foreach ($branch in $branches)
     continue
   }
 
+  if ($LastCommitOlderThan) {
+    if (!$res.object -or !$res.object.url) {
+      LogDebug "No commit url returned from response. Skipping... "
+      continue
+    }
+    $commitDate = GetCommitDate -commitUrl $res.object.url
+    if ($commitDate -gt $LastCommitOlderThan) {
+      LogDebug "The branch $branch last commit date $commitDate is newer than the date $LastCommitOlderThan. Skipping."
+      continue
+    }
+  } 
   LogDebug "Branch [ $branchName ] in repo [ $RepoName ] has no associated open Pull Request. Deleting Branch"
   try{
-    Remove-GitHubSourceReferences -RepoOwner $RepoOwner -RepoName $RepoName -Ref ($branch.Remove(0,5)) -AuthToken $AuthToken
+    Remove-GitHubSourceReferences -RepoNameWithOwner $RepoNameWithOwner -Ref ($branch.Remove(0,5)) -AuthToken $AuthToken
```

Comment:
Lets not do the Remove(0,5) here and just pass the branch name in and if we need to cleanup the "refs" part do it in that function.

Reply:
The $ref in `Remove-GitHubSourceReference` can be branches/tags etc. 
For branch, they are taking "heads/$branchName". So it is better to have the regex to work on the branch.

Reply:
If we only remove "refs/" from the passed in branch in that function it shouldn't impact other refs.

Reply:
The API only accept "heads/$branchName". For the format of `refs/heads/$branchName`, `heads/$branchName` or `$branchName`, if we allow to handle two of them, then we'd better to handle the rest as well. Otherwise, we only allow to pass the format which API is taking. 

Then, we have to add "heads/" in front of "$branchName", if they pass "$branchName" only.